Add WEBrick dependency

voxik opened the pull request:
Because WEBrick was removed from Ruby 3.0.

voxik commented:
And there is one more issue:

~~~
+ rspec spec
#<Thread:0x000055583c885240 /builddir/build/BUILD/typhoeus-1.4.0/usr/share/gems/gems/typhoeus-1.4.0/spec/support/localhost_server.rb:75 run> terminated with exception (report_on_exception is true):
/usr/share/gems/gems/rack-2.2.3/lib/rack/handler/webrick.rb:26:in `run': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from /builddir/build/BUILD/typhoeus-1.4.0/usr/share/gems/gems/typhoeus-1.4.0/spec/support/localhost_server.rb:46:in `boot'
	from /builddir/build/BUILD/typhoeus-1.4.0/usr/share/gems/gems/typhoeus-1.4.0/spec/support/localhost_server.rb:29:in `block in initialize'
	from /builddir/build/BUILD/typhoeus-1.4.0/usr/share/gems/gems/typhoeus-1.4.0/spec/support/localhost_server.rb:75:in `block in concurrently'
~~~

Not really sure if the fix is compatible with older Ruby.

voxik commented:
Apparently, this breaks Ruby 1.9.3 and older, but does anybody care these days? Even on RHEL7, there is at least Ruby 2.0.0.

Kjarrigan commented:
In the line just below where you added the webrick gem there is an if clause that checks for `Ruby >= 1.9.0` Can you please add such a wrapper around webrick for >= 2.0. Although I agree that Ruby 1.9.3 is already EOL for a while it is an easy "fix" to keep the compatibility.

voxik commented:
I have moved WEBrick down to the section you have referred to (although the issues is not WEBrick incompatibility) and added version guard around the call. Lets see if it helps to keep the compatibility.

voxik commented:
Hm, it is reported by parser as syntax error :angry: Not really sure how to deal with this in backward compatible manner

felipedmesquita commented:
Hi @voxik, thanks for the pr. The conditional dependency on WEBrick has been added in 8eac36b for ruby 3:
```ruby
 if Gem.ruby_version >= Gem::Version.new("3.0.0")
    gem 'webrick'
  end
```
Since this change already done, I'm closing this pr. Please open a new one if you run into other issues or find the current solution insufficient.
